### DISCUSSION TITLE
Fetching all one-time donations

### DISCUSSION THREAD

**Discussion by nzakas** [opencollective/opencollective#5226] [Development Q&A] [answered]
Currently, to download one-time donations, I'm using the GraphQL API to retrieve `transactions` and then I filter on anything that has `order.frequency` of `ONETIME`. While this works, it also requires me to download a lot of other data that I throw away, meaning I end up needing to do five different requests just to end up with around 20 one-time donations (because repeated donations are more common than one-time donations). 

I'm curious if there's an easier way to download one-time donations? If not, would it be possible to add a filter onto `transactions` that would allow me to return just one-time donations?

Thanks!

**Comment by znarf** [comment]
What you propose could work but I think a more suitable approach is to have that filter on the `orders` directly.

I started implement something like this here: https://github.com/opencollective/opencollective-api/pull/7178

**Comment by znarf** [comment, marked as answer]
Shipped. You can have a try with the following:

```graphql
query {
  orders(
    account: { slug: "eslint" }
    frequency: ONETIME
    status: PAID
    limit: 1000
  ) {
    totalCount
    nodes {
      frequency
      status
      amount {
        value
        currency
      }
      fromAccount {
        slug
      }
    }
  }
}

```

**Reply by nzakas** [reply]
Thank you! 🙏  I will give this a try next week and report back.

**Reply by nzakas** [reply]
This worked brilliantly. Thanks so much for the fast turnaround.
